src/cljam/io/vcf/util/normalize.clj

```diff
@@ -0,0 +1,163 @@
+(ns cljam.io.vcf.util.normalize
+  (:require [clojure.string :as cstr]
+            [cljam.io.vcf.util :as vcf-util]
+            [cljam.io.sequence :as io-seq]))
+
+(defn- biallelic-map
+  [m id->number ^long i ^long n-allele]
+  (let [ploidy (or (some-> m :GT vcf-util/parse-genotype count) 2)]
+    (into
+     {}
+     (map
+      (fn [[k v]]
+        (let [n (id->number k)]
+          [k (cond
+               (= k :GT) (vcf-util/biallelic-genotype v (inc i))
+               (= n "A") [(nth v i)]
+               (= n "R") [(first v) (nth v (inc i))]
+               (= n "G") (vcf-util/biallelic-coll ploidy n-allele (inc i) v)
+               :else v)])))
+     m)))
+
+(defn- get-id->number
+  [coll]
+  (into {} (map (juxt (comp keyword :id) :number)) coll))
+
+(defn make-splitter
+  "Returns a function that splits a multiallelic variant into a sequence of
+  biallelic variants."
+  [{:keys [info format]} header]
+  (let [info-id->number (get-id->number info)
+        format-id->number (get-id->number format)
+        samples (map keyword (drop 9 header))]
+    (fn split [{:keys [alt] :as v}]
+      (let [n (count alt)]
+        (if (<= n 1)
+          [v]
+          (->> alt
+               (map-indexed
+                (fn sample-ith-alt [i allele]
+                  (-> v
+                      (assoc :alt [allele])
+                      (update :info biallelic-map info-id->number i n)
+                      (into
+                       (map
+                        (fn [s]
+                          [s (biallelic-map (v s) format-id->number i n)]))
+                       samples))))))))))
+
+(defn same-ref?
```

Comment:
This function doesn't seem to be used anywhere in this namespace. Is it necessary to be defined here?

Reply:
No, not necessary. Do you mean it should be in `cljam.io.vcf.util`?

Reply:
Yeah, I think it would suit better to `cljam.io.vcf.util` if it is not specific to VCF normalization.

Reply:
Considering that current `cljam.io.vcf.util` is a namespace for VCF/BCF and that `same-ref?` compares two files of different types, how about making a new namespace like `cljam.io.vcf.util.check`?

Reply:
That sounds good, too 👍